nixos/modules/services/web-apps/nextcloud.nix

```diff
@@ -468,10 +458,18 @@ in {
           script = ''
             chmod og+x ${cfg.home}
             ln -sf ${cfg.package}/apps ${cfg.home}/
-            mkdir -p ${cfg.home}/config ${cfg.home}/data ${cfg.home}/store-apps
-            ln -sf ${overrideConfig} ${cfg.home}/config/override.config.php
 
-            chown -R nextcloud:nginx ${cfg.home}/config ${cfg.home}/data ${cfg.home}/store-apps
+            # create nextcloud directories.
+            # if the directories exist already with wrong permissions, we fix that
+            for dir in ${cfg.home}/config ${cfg.home}/data ${cfg.home}/store-apps; do
+              if [ ! -e $dir ]; then
+                install -o nextcloud -g nextcloud -d $dir
+              elif [ $(stat -c "%G" $dir) != "nextcloud" ]; then
+                chown -R nextcloud:nextcloud $dir
```

Comment:
This script is run as `nextcloud` so this `chown` doesn't make sense to me. The code this script replaces also had that, though... can someone explain this to me?

Reply:
The chown command is only for fixing the group permissions for installation prior to this PR.
Those installations will have `nextcloud:nginx` and it will be changed to `nextcloud:nextcloud`. This command is possible as nextcloud user.

Other reasons why I wrote the script like this:
1. We can't just do a plain `install -o nextcloud -g nextcloud -d ${cfg.home}/config ${cfg.home}/data ${cfg.home}/store-apps`. Because if the these dirs already exists with wrong group, it will only fix the group for the top level dirs, but not their contents.
1. We also don't want to do an unconditional `chown -R` on every single startup, since this is an expensive operation and might not even work depending on the fs.

As of this script, it will only create the top level dirs if they don't exist and it will only fix the group if the group is wrong.

Reply:
That makes sense to me. Maybe consider `chgrp` if the intention is to only change the group.